homeassistant/components/guardian/binary_sensor.py

```diff
@@ -1,66 +1,153 @@
-    DATA_CLIENT,
+    CONF_UID,
     DATA_COORDINATOR,
     DOMAIN,
+    SIGNAL_PAIRED_SENSOR_COORDINATOR_ADDED,
 )
 
 ATTR_CONNECTED_CLIENTS = "connected_clients"
 
 SENSOR_KIND_AP_INFO = "ap_enabled"
 SENSOR_KIND_LEAK_DETECTED = "leak_detected"
-SENSORS = [
-    (SENSOR_KIND_AP_INFO, "Onboard AP Enabled", "connectivity"),
-    (SENSOR_KIND_LEAK_DETECTED, "Leak Detected", "moisture"),
-]
+SENSOR_KIND_MOVED = "moved"
+
+SENSOR_ATTRS_MAP = {
+    SENSOR_KIND_AP_INFO: ("Onboard AP Enabled", "connectivity"),
+    SENSOR_KIND_LEAK_DETECTED: ("Leak Detected", "moisture"),
+    SENSOR_KIND_MOVED: ("Recently Moved", "moving"),
+}
+
+PAIRED_SENSOR_SENSORS = [SENSOR_KIND_LEAK_DETECTED, SENSOR_KIND_MOVED]
+VALVE_CONTROLLER_SENSORS = [SENSOR_KIND_AP_INFO, SENSOR_KIND_LEAK_DETECTED]
 
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: Callable
 ) -> None:
     """Set up Guardian switches based on a config entry."""
-    async_add_entities(
-        [
-            GuardianBinarySensor(
+
+    async def add_new_paired_sensor(uid: str) -> None:
+        """Add a new paired sensor."""
+        coordinator = hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id][
+            API_SENSOR_PAIRED_SENSOR_STATUS
+        ][uid]
+
+        entities = []
+        for kind in PAIRED_SENSOR_SENSORS:
+            name, device_class = SENSOR_ATTRS_MAP[kind]
+            entities.append(
+                PairedSensorBinarySensor(
+                    entry, coordinator, kind, name, device_class, None,
+                )
+            )
+
+        async_add_entities(entities, True)
+
+    # Handle adding paired sensors after HASS startup:
+    async_register_dispatcher_connect(
+        hass,
+        entry,
+        SIGNAL_PAIRED_SENSOR_COORDINATOR_ADDED.format(entry.data[CONF_UID]),
+        add_new_paired_sensor,
+    )
+
+    sensors = []
+
+    # Add all valve controller-specific binary sensors:
+    for kind in VALVE_CONTROLLER_SENSORS:
+        name, device_class = SENSOR_ATTRS_MAP[kind]
+        sensors.append(
+            ValveControllerBinarySensor(
                 entry,
-                hass.data[DOMAIN][DATA_CLIENT][entry.entry_id],
                 hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id],
                 kind,
                 name,
                 device_class,
+                None,
             )
-            for kind, name, device_class in SENSORS
-        ],
-        True,
-    )
+        )
+
+    # Add all paired sensor-specific binary sensors:
+    for coordinator in hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id][
+        API_SENSOR_PAIRED_SENSOR_STATUS
+    ].values():
+        for kind in PAIRED_SENSOR_SENSORS:
+            name, device_class = SENSOR_ATTRS_MAP[kind]
+            sensors.append(
+                PairedSensorBinarySensor(
+                    entry, coordinator, kind, name, device_class, None,
+                )
+            )
+
+    async_add_entities(sensors, True)
```

Comment:
Why pass `True` ? They come from an update coordinator that already has the latest data ?